src/app/admin/email-templates/email-template-form.component.ts

```diff
@@ -79,21 +92,62 @@ export class AdminEmailTemplateFormComponent extends FormComponent<EmailTemplate
     return this._ckEditorConfig;
   }
 
+  @ViewChild(CKEditorComponent, { static: false })
+  public set ckEditor(value: CKEditorComponent) {
+    this._ckEditor = value;
+  }
+
   constructor(service: AdminEmailTemplateService,
               router: Router,
               route: ActivatedRoute,
               notificationService: NotificationService,
               breadcrumbService: BreadcrumbService,
-              protected _translateService: TranslateService) {
+              protected _translateService: TranslateService,
+              protected _cdr: ChangeDetectorRef) {
     super(service, router, route, notificationService, breadcrumbService);
   }
 
   public ngOnInit() {
     this._ckEditorConfig.language = this._translateService.currentLang;
+
+    this._onLanguageChange();
+
     super.ngOnInit();
   }
 
+  public ngOnDestroy() {
+    this._destroy$.next(true);
+    this._destroy$.complete();
+  }
+
   public addText(editorArea: CKEditorComponent, text: string) {
     editorArea.instance.insertText(`{{ ' ${text} ' }}`);
   }
+
+  protected _removeCurrentEditor(): void {
+    this._ckEditor.instance.removeAllListeners();
+
+    window['CKEDITOR'].instances[this._ckEditor.instance.name].destroy();
+    document.getElementsByClassName(this._ckEditor.instance.id).item(0).remove();
```

Comment:
it will be better to use straight `document.getElementById('editor')`

Reply:
This is line is removing original CkEditor instance from DOM (please don't confuse with Angular component), and in there it represents as:
`div id="cke_${_this._ckEditor.instance.name}" class="${this._ckEditor.instance.id}..."`

I implemented my first solution as you suggest but after thought about if someone change `cke_` prefix by any reason I rewrote solution to looking up by plain className that stored in JS as `id` property.

What do you think should I return back to my first solution or not?

Reply:
I was thinking about re-instantiation of the whole `<ckeditor>` not the actual WYSIWYG editor itself. In the template it has `id='editor'` and can be retrieved by it.

Reply:
Yep, is another variant of solution. 

I'll pick an additional time to provide more appropriate solution, because all these solutions have cons in share responsibilities in a component that shouldn't know about destroying CKEditor anything.

Reply:
As a result, was implemented a solution with dynamically changing editors by the picked language via structural directives (ngFor + ngIf)